instrumentation/opentelemetry-instrumentation-psycopg2/src/opentelemetry/instrumentation/psycopg2/__init__.py

```diff
@@ -158,33 +158,40 @@ def _uninstrument(self, **kwargs):
         dbapi.unwrap_connect(psycopg2, "connect")
 
     # TODO(owais): check if core dbapi can do this for all dbapi implementations e.g, pymysql and mysql
-    @staticmethod
-    def instrument_connection(connection, tracer_provider=None):
-        if not hasattr(connection, "_is_instrumented_by_opentelemetry"):
```

Comment:
I don't think this is an actual solution to the issue. We have this logic here to prevent double instrumentation of the same `connection` object. @qiuge615 Are you looking to simply unblock yourself for now or are you looking to find an actual fix for why `connection` is throwing exceptions when being assigned properties?

Reply:
@lzchen Thanks for your comments. I updated the codes with previous logic, use DatabaseApiIntegration to instrument connection and use "isinstance(connection, ObjectProxy)" to check if the connection already instrumented, at last use dbapi to create TracedConnectionProxy. Tested locally, it can instrument connection and display warning message if double instrumentation. If this approach makes sense to you, I will update PR accordingly, thanks.

Reply:
@qiuge615 

Did you update the code or are you planning to update the code because I don't see any new commit.

Reply:
I did not push the code yet. I will push the changes for review. In the meanwhile if you have any suggestions or comments, please let me know, I will update accordingly, thank you.

Reply:
@qiuge615 

It will be easier to make suggestions + comments once the changes are pushed for review thanks.

Reply:
@lzchen I updated the code, please review and share your comments, thanks.